scylla/src/transport/session_test.rs

```diff
@@ -75,6 +75,14 @@ async fn test_unprepared_statement() {
             (7, 11, &String::from(""))
         ]
     );
+    let query_result = session
+        .query_iter("SELECT a, b, c FROM ks.t", &[])
+        .await
+        .unwrap();
+    for (spec, name) in query_result.get_column_specs().iter().zip(["a", "b", "c"]) {
+        assert_eq!(spec.name, name); // Check column name.
+        assert_eq!(spec.table_spec.ks_name, "ks");
+    }
```

Comment:
This test doesn't detect the case when the `get_column_specs` is empty. The `zip` method takes two iterators and produces a new one which stops after the shorter of them finishes ([doc link](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip)) - so if the columns specs slice is empty, the loop will immediately stop.

Besides, it's precisely what happens now. The way that the `RowIterator` is implemented now is that it caches a single page of rows and lazily updates it after the page is exhausted and the `next` method is called - but the iterator starts with a dummy, empty page (see e.g. [here](https://github.com/scylladb/scylla-rust-driver/blob/83199faac35accfd18910d9b5fa7cf5f08384248/scylla/src/transport/iterator.rs#L151)) which has empty metadata. So, until you call `next()` on the iterator for the first time, you will get empty metadata.

To solve this problem, let's change the `Session::execute_iter` and `Session::query_iter` so that they return an iterator already initialized with the first page. Most of the changes needed to implement this should happen in `RowIterator::new_for_query` and `RowIterator::new_for_prepared_statement`.